### PR TITLE
tools/downloader: separate download engine

### PR DESCRIPTION
**Pull request by jkamelin**
Separates download engine into python module. `download_engine` module consists of downloader, cache, postprocessing and file source-related submodules.

### CODE REVIEW

File: tools/downloader/src/open_model_zoo/model_tools/download_engine/cache.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        with open(cache_path, 'rb') as cache_file, open(path, 'wb') as destination_file:
+            while True:
+                data = cache_file.read(CHUNK_SIZE)
+                if not data:
+                    break
+                cache_size += len(data)
+                if cache_size > model_file.size:
+                    reporter.log_error("Cached file is longer than expected ({} B), copying aborted", model_file.size)
+                    return False
+                cache_sha256.update(data)
```
Comment (SDxKeeper) [inline tools/downloader/src/open_model_zoo/model_tools/download_engine/cache.py:64]: side comment: did we test if using shutil.copyfile and reading destination file for sha256 calculation is slower that doing both in one shot like in this implementation?

File: tools/downloader/src/open_model_zoo/model_tools/_configuration.py [hunk 2]
```diff
@@ -13,155 +13,20 @@
...
-import shutil
 import sys
-
-from pathlib import Path
-
-import requests
 import yaml
 
 from open_model_zoo.model_tools import _common
+from open_model_zoo.model_tools.download_engine import file_source, postprocessing, validation
```
Comment (SDxKeeper) [inline tools/downloader/src/open_model_zoo/model_tools/_configuration.py:23]: Looks like FileSource is enough here, but whole module is imported. Is it to state module attachment in the code?

File: tools/downloader/src/open_model_zoo/model_tools/download_engine/file_source.py [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import requests
+
+from open_model_zoo.model_tools.download_engine import validation
+
+DOWNLOAD_TIMEOUT = 5 * 60
```
Comment (SDxKeeper) [inline tools/downloader/src/open_model_zoo/model_tools/download_engine/file_source.py:20]: I think this is something user shall be able to control.

**Review by SDxKeeper — APPROVED** [review]

File: tools/downloader/src/open_model_zoo/model_tools/download_engine/file_source.py (hunk 3 above)
Comment (jkamelin) [reply tools/downloader/src/open_model_zoo/model_tools/download_engine/file_source.py]: added as a argument to `Downloader`

**Comment by jkamelin** [comment]
@Wovchena, please review

**Review by icvgh — APPROVED** [review]

**Review by Wovchena — APPROVED** [review]

**Pull request merged**